File: livingrimoire start here/LivinGrimoire java/LivinGrimoire/Brain.java
package LivinGrimoire;

public class Brain {
    // c'tor
    private String emotion = "";
    private String logicLobeOutput = "";
    public Lobe logicLobe = new Lobe();
    public Lobe hardwareLobe = new Lobe();
    public Lobe ear = new Lobe();
    public Lobe skin = new Lobe();
    public Lobe eye = new Lobe();

    public Brain() {
        Brain.imprintSoul(logicLobe.getKokoro(), hardwareLobe, ear, skin, eye);
    }

    public static void imprintSoul(Kokoro kokoro, Lobe... args) {
        for (Lobe arg : args) { arg.setKokoro(kokoro); }
    }

    // ret active alg part representing emotion
    public String getEmotion() {
        return emotion;
    }

    // ret feedback (last output)
    public String getLogicLobeOutput() {
        return logicLobeOutput;
    }

    public void setDatabase(AbsDictionaryDB db) {
        // sets same database to all lobes
        logicLobe.setDatabase(db);
    }

    // live
    public void doIt(String ear, String skin, String eye) {
        logicLobeOutput = logicLobe.think(ear, skin, eye);
        emotion = logicLobe.getSoulEmotion();
        hardwareLobe.think(logicLobeOutput, skin, eye);
    }

    public void addSkill(Skill skill) {
        // Adds a skill to the correct Lobe based on its skillLobe attribute.
        // Just pass the skill—the Brain handles where it belongs.
        switch (skill.getSkillLobe()) {
            case 1: logicLobe.addSkill(skill); break;    // Logical skill
            case 2: hardwareLobe.addSkill(skill); break; // Hardware skill
            case 3: ear.addSkill(skill); break;          // Ear skill
            case 4: skin.addSkill(skill); break;         // Skin skill
            case 5: eye.addSkill(skill); break;          // Eye skill
        }
    }

    public void removeSkill(Skill skill) {
        // Removes a skill from the correct Lobe based on its skillLobe attribute.
        // Just pass the skill—the Brain handles its removal.
        switch (skill.getSkillLobe()) {
            case 1: logicLobe.removeSkill(skill); break;
            case 2: hardwareLobe.removeSkill(skill); break;
            case 3: ear.removeSkill(skill); break;
            case 4: skin.removeSkill(skill); break;
            case 5: eye.removeSkill(skill); break;
        }
    }

    public Brain chained(Skill skill) {
        // chained add skill
        addSkill(skill);
        return this;
    }

    // add regular thinking(logical) skill
    public void addLogicalSkill(Skill skill) {
        logicLobe.addRegularSkill(skill);
    }

    // add output skill
    public void addHardwareSkill(Skill skill) {
        hardwareLobe.addRegularSkill(skill);
    }

    // add audio(ear) input skill
    public void addEarSkill(Skill skill) {
        ear.addRegularSkill(skill);
    }

    // add sensor input skill
    public void addSkinSkill(Skill skill) {
        skin.addRegularSkill(skill);
    }

    // add visual input skill
    public void addEyeSkill(Skill skill) {
        eye.addRegularSkill(skill);
    }

    public void think(String keyIn) {
        if (keyIn != null && !keyIn.isEmpty()) {
            // handles typed inputs(keyIn)
            doIt(keyIn, "", ""); // the string is not empty
        } else {
            // the string is empty, process with sensory inputs
            doIt(ear.think("", "", ""), skin.think("", "", ""), eye.think("", "", ""));
        }
    }

    public void think() {
        doIt(ear.think("", "", ""), skin.think("", "", ""), eye.think("", "", ""));
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/livinGrimoire/Brain.kt
package livinGrimoire

class Brain {
    var logicChobit = Chobits()

    // ret active alg part representing emotion
    var emotion = ""
        private set

    // ret feedback (last output)
    var logicChobitOutput = ""
        private set
    var hardwareChobit = Chobits()
    var ear = Chobits()
    var skin = Chobits()
    var eye = Chobits()

    //c'tor
    init {
        imprintSoul(logicChobit.kokoro, hardwareChobit, ear, skin, eye)
    }

    // live
    fun doIt(ear: String, skin: String, eye: String) {
        logicChobitOutput = logicChobit.think(ear, skin, eye)
        emotion = logicChobit.getSoulEmotion()
        hardwareChobit.think(logicChobitOutput, skin, eye)
    }

    fun addSkill(skill: Skill) {
        /*
        Adds a skill to the correct Chobits based on its skill_lobe attribute.
        Just pass the skill—the Brain handles where it belongs.
        */
        when (skill.skillLobe) {
            1 -> logicChobit.addSkill(skill)
            2 -> hardwareChobit.addSkill(skill)
            3 -> ear.addSkill(skill)
            4 -> skin.addSkill(skill)
            5 -> eye.addSkill(skill)
        }
    }

    fun chained(skill: Skill): Brain {
        // chained add skill
        addSkill(skill)
        return this
    }

    // add regular thinking(logical) skill
    fun addLogicalSkill(skill: Skill) {
        logicChobit.addRegularSkill(skill)
    }

    // add output skill
    fun addHardwareSkill(skill: Skill) {
        hardwareChobit.addRegularSkill(skill)
    }

    // add audio(ear) input skill
    fun addEarSkill(skill: Skill) {
        ear.addRegularSkill(skill)
    }

    // add sensor input skill
    fun addSkinSkill(skill: Skill) {
        skin.addRegularSkill(skill)
    }

    // add visual input skill
    fun addEyeSkill(skill: Skill) {
        eye.addRegularSkill(skill)
    }

    fun think(keyIn: String) {
        if (!keyIn.isEmpty()) {
            // handles typed inputs(keyIn)
            doIt(keyIn, "", "")
        } else {
            // accounts for sensory inputs
            doIt(ear.think("", "", ""), skin.think("", "", ""), eye.think("", "", ""))
        }
    }

    fun think() {
        // accounts for sensory inputs only. use this overload for tick events(where it is certain no typed inputs are to be processed)
        doIt(ear.think("", "", ""), skin.think("", "", ""), eye.think("", "", ""))
    }

    companion object {
        private fun imprintSoul(kokoro: Kokoro, vararg args: Chobits) {
            for (arg in args) {
                arg.kokoro = kokoro
            }
        }
    }
}